### ISSUE TITLE
Re-sharing error message shows LDAP uuid instead of displayname

### ISSUE THREAD

**Issue by gig13** [owncloud/core#24681]
### Steps to reproduce
1. Setup ownCloud 8.2.5 with LDAP
2. Try to re-share a file within a folder that had already been shared with them
3.

### Expected behaviour
The pop up message should show the displayname when stating the file has already been shared

### Actual behaviour
The message shows the uuid instead of the displayname

### Server configuration
**Operating system**: RHEL

**Web server:** Apache

**Database:** MySQL

**PHP version:** 5.5

**ownCloud version:** 8.2.5

**Updated from an older ownCloud or fresh install:** Updated

**Where did you install ownCloud from:** archives

Screen shot
![ldap uuid message](https://cloud.githubusercontent.com/assets/4803385/15334773/963764d2-1c3d-11e6-9f6e-2f0c993de9bd.png)

@bboule @PVince81

**Comment by PVince81** [comment]
@rullzer @blizzz @davicente any clues ?

Maybe need to explicitly convert the user id before adding it to error message ?
Ideally we should use the display name instead.

**Comment by rullzer** [comment]
This is most likely burried in one of the manny error messages in the ancient sharing code ;)

**Comment by gig13** [comment]
@rullzer @PVince81  Any updates to provide the customer?

**Comment by davicente** [comment]
@gig13 I cannot reproduce it. 
1) I share a folder with an user (aaliyah hayes)
2) I try to share a file within the previous folder to the same user

What happens:
The user doesn't appear to be shared, so I can't share this file with him

Sharing folder
![sharing_1](https://cloud.githubusercontent.com/assets/752240/15498470/09cba504-21a0-11e6-9efa-7f4d0a7060ea.png)

Trying to share the file within the folder
![sharing_2](https://cloud.githubusercontent.com/assets/752240/15498469/09c7167e-21a0-11e6-83c0-044846211894.png)

Could you give me some more information about the steps to reproduce it? Here or in #owncloud-dev ( https://webchat.freenode.net/?channels=owncloud-dev ) (IRC)

**Comment by gig13** [comment]
@davicente I will try and get more detailed reproduction steps

**Comment by gig13** [comment]
1. User A shares a folder with Users B and C. There are some files in the folder.
2. User B tries to share a file within that folder with User C. That should trigger the error message.

**Comment by cdamken** [comment]
> @rullzer @blizzz @davicente any clues ?
> 
> Maybe need to explicitly convert the user id before adding it to error message ?
> Ideally we should use the display name instead.

I'm seeing that the configuration from LDAP is:

```
"s01ldap_expert_username_attr": "",
"s02ldap_expert_username_attr": "",
```

That's why the default username-ID is UUID when the users are over LDAP and not modified.

A solution could be use the `Display Name` but not all back-ends does have it. 

The other  possibility could be to change the output of the message.

**Comment by cdamken** [comment]
Quoting @scolebrook 

In iuser.php under /lib/public

/**
* get the display name for the user, if no specific display name is set it will fallback to the user id
*
* @return string
* @since 8.0.0
*/
public function getDisplayName();

So all references objects that implement IUser should be using getDisplayName for messages like this which will return the appropriate name with the appropriate fall backs to cover all possible configurations and backends. And if this message is getting uid without referring to an object that implements IUser, then that is itself a bug.

**Comment by scolebrook** [comment]
After doing some digging the issue is caused in /lib/private/share/share.php. In the shareItem function it has 4 error messages which refer to the $shareWith variable that was passed to it, in this instance by /core/ajax/share.php. $shareWith will be the internal username of the recipient user.

Instead of $shareWith being supplied as input for these error messages, \OC_User::getDisplayName($shareWith) should probably be used. It will return the display name if the user backend provides one or fall back to returning the value in $shareWith. Given this is in private, I'm not sure if using \OCP\User::getDisplayName() would be preferred or not. It just calls the OC_User function of the same name.

**Comment by PVince81** [comment]
In OC >= 9.0 the message was replaced with a yellow banned "Path is already shared with that user" without names

**Comment by scolebrook** [comment]
@PVince81 I still see the same code in master (although the case of the file name has been changed). Rather than display the message from the Exception, 9.0+ displays it's own message in these circumstances? The Exception message should probably still be fixed. The log form of the message can be left to display username/guid. That's appropriate in a log file I guess.

If you can let me know whether I should use \OCP\User or \OC_User I can put a PR together. Let me know which branch if not master.

**Comment by PVince81** [comment]
@scolebrook yeah, there is some old code from ajax/share that hasn't been deleted yet for legacy reasons, but the web UI doesn't use it any more.

If you want to fix the message for 8.2, the branch is stable8.2. I think using `\OCP\User::getDisplayName($shareWith)` should be fine for that version.
For performance reasons, I think you should try doing it just before showing the message and not in advance because that might trigger some database query.

**Comment by scolebrook** [comment]
https://github.com/owncloud/core/pull/25305
Tested on 8.2.5 successfully.

**Comment by PVince81** [comment]
Closing as fixed in 8.2.7
